allennlp/training/metrics/metric.py

```diff
@@ -50,9 +51,11 @@ def from_params(cls, params: Params, vocab: Optional[Vocabulary] = None):
     @staticmethod
     def unwrap_to_tensors(*tensors):
         """
-        If you actually passed in Variables to a Metric instead of Tensors, there will be
+        If you actually passed gradient-tracking Tensors to a Metric, there will be
         a huge memory leak, because it will prevent garbage collection for the computation
         graph. This method ensures that you're using tensors directly and that they are on
         the CPU.
         """
-        return (x.data.cpu() if isinstance(x, torch.autograd.Variable) else x for x in tensors)
+        # pylint: disable=unidiomatic-typecheck
+        return (x.data.cpu() if is_tensor(x) else x
```

Comment:
One line?

Reply:
And probably just fail if you pass something that's not a tensor, instead of checking `is_tensor`.

Reply:
`.detach()` instead of `.data`

Reply:
removing the `else x` behavior breaks a lot of tests (not sure if it's just how the tests are written or if it actually breaks our code)

Reply:
I just checked this, and it's because we pass in masks that could be `None`.  It might be worth changing this to `(x.data.cpu() if x is not None else None for x in tensors)`, just to fail explicitly instead of silently if you're passing something in that we're not expecting.  Not a big deal, though.